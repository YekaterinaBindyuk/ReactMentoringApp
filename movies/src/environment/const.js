export const BASE_IMG_CLASSNAME = 'img';
export const MOVIE_LIST_ITEM_COVER = 'small';
export const MOVIE_COVER = 'large';
export const MAIN_LOGO = 'http://localhost:8080/src/images/logo.png';
export const SEARCH_BUTTON = 'Search';
export const TITLE_TYPE_BUTTON = 'title';
export const GENRE_TYPE_BUTTON = 'genre';
export const WARNING = 'Something went wrong';
export const RATING_TYPE_BUTTON = 'rating';
export const RELEASE_TYPE_BUTTON = 'release date';
export const SORTING_TEXT = 'Sort by: ';
export const SEARCH_TEXT = 'Search by: ';
export const NO_FILMS_WARNING = 'No films found';
export const MOVIES_COUNT_TEXT = ' films found:';
export const RECOMMENDED_MOVIES = 'Recommended movies';
export const MAIN_PAGE = '/';
export const REST_URL = 'https://reactjs-cdp.herokuapp.com/';
export const MOVIES_URL = REST_URL + 'movies';
