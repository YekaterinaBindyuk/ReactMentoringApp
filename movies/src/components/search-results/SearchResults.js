import React from "react";
import SearchResultsCount from "./SearchResultsCount";
import MoviesList from "../movies-list/MoviesList";
import Sorting from "../sorting/Sorting";
import { NO_FILMS_WARNING } from "../../environment/const";
import { connect } from "react-redux";

import "../../style/moviesList.css";

export class SearchResults extends React.Component {
  render = () => {
    const { moviesList } = this.props;
    return (
      <main>
        {moviesList.length ? (
          <div>
            <div className="search-results-info">
              <SearchResultsCount count={moviesList.length} />
              <Sorting />
            </div>
            <MoviesList moviesList={moviesList} />
          </div>
        ) : (
          <div>{NO_FILMS_WARNING}</div>
        )}
      </main>
    );
  };
}


const mapStateToProps = state => {
  return {
    moviesList: state.movies
  };
};

export default connect(
  mapStateToProps,
  null
)(SearchResults);
