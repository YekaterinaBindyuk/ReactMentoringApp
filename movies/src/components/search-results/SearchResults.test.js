import React from "react";
import { render, shallow } from "enzyme";
import { NO_FILMS_WARNING } from "../../environment/const";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";

import { SearchResults } from "./SearchResults";
describe("SearchResults", () => {
  it('should render correctly in "debug" mode', () => {
    const component = render(<SearchResults debug moviesList={[]} />);
    expect(component).toMatchSnapshot();
  });
});

it("renders no films warning in case of absense of films", () => {
  const mockGetMoviesfn = jest.fn();
  const wrapper = shallow(
    <SearchResults moviesList={[]} getMovies={mockGetMoviesfn} />
  );
  expect(wrapper.html()).toEqual(
    "<main><div>" + NO_FILMS_WARNING + "</div></main>"
  );
});

it("renders films", () => {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const moviesList = [
    {
      budget: 0,
      genres: ["Comedy"],
      id: 447365,
      overview: "The third film based on Marvel's Guardians of the Galaxy.",
      poster_path:
        "https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg",
      release_date: "2020-05-01",
      revenue: 0,
      runtime: null,
      tagline: "",
      title: "Guardians of the Galaxy Vol. 3",
      vote_average: 0,
      vote_count: 9
    }
  ];
  const store = mockStore({ movies: moviesList, movie: {}, recommendedMovies: [] });
  const mockGetMoviesfn = jest.fn();
  
  const wrapper = shallow(
    <Provider store={store}>
      <SearchResults moviesList={moviesList} getMovies={mockGetMoviesfn} />
    </Provider>
  );
  expect(wrapper.html()).toEqual(

'<main><div><div class="search-results-info"><div class="movies-count"><div class="badge badge-dark p-2"><i class="fa"></i><span> films found: <span class="badge badge-light">1</span></span></div></div><div class="sort">Sort by: <button id="sort-by-release-btn" class="btn btn-dark m-1"><i class="fa"></i>release date</button><button id="sort-by-rating-btn" class="btn btn-dark"><i class="fa"></i><i class="fa"></i><i class="fa"></i>rating</button></div></div><div class="movies-list"><div class="movies-list-item"><img src="https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg" alt="Guardians of the Galaxy Vol. 3" class="img_small"/><h3 class="title">Guardians of the Galaxy Vol. 3</h3><section><div class="genres">Comedy</div></section><div>2020-05-01</div></div></div></div></main>'
    );
});
