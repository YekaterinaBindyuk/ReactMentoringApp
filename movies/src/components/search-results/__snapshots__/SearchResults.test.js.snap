// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults should render correctly in "debug" mode 1`] = `
<main>
  <div>
    No films found
  </div>
</main>
`;
